### PR TITLE
Create with-firebase-hosting-and-docker example

### PR DESCRIPTION
**Pull request by sarovin**
Adds an example based off of @jthegedus work on firebase hosting, compatible with next v7 and using docker for deploy, dev and serve.

### CODE REVIEW

File: examples/with-firebase-hosting-and-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+### Firebase deploy
+
+```bash
+yarn docker:deploy
+```
+
+## The idea behind the example
+
+The goal is to host the Next.js app on Firebase Cloud Functions with Firebase Hosting rewrite rules so our app is served from our Firebase Hosting URL, with docker support. Each individual `page` bundle is served in a new call to the Cloud Function which performs the initial server render.
```
Comment (jthegedus) [inline examples/with-firebase-hosting-and-docker/README.md]: I think this example should more clearly explain that the use of Docker is entirely for local development and the deployment process and that Firebase itself will not leverage Docker itself.

**Comment by sarovin** [comment]
@jthegedus What do you think about the new README?

**Review by jthegedus — APPROVED** [review]
:100:

**Comment by timneutkens** [comment]
Thank you @sarovin!

@jthegedus thanks for reviewing 💯

**Pull request merged**